tests/aodvv2/aodv_writer_tests.c

```diff
@@ -0,0 +1,178 @@
+        .timestamp = now,
+    };
+
+    vtimer_now(&now);
+    plain_rrep = (struct aodvv2_packet_data) {
+        .hoplimit = AODVV2_MAX_HOPCOUNT,
+        .sender = aodvv2_test_sender_ta,
+        .metricType = AODVV2_DEFAULT_METRIC_TYPE,
+        .origNode = {
+            .addr = aodvv2_test_origaddr,
+            .metric = 4,
+            .seqnum = 1,
+        },
+        .targNode = {
+            .addr = aodvv2_test_targaddr,
+            .metric = 2,
+            .seqnum = 2,
+        },
+        .timestamp = now,
+    };
+
+    return 0;
+}
+
+static void aodv_test_write_packet(struct rfc5444_writer *wr __attribute__ ((unused)),
+                          struct rfc5444_writer_target *iface __attribute__((unused)),
+                          void *buffer, size_t length)
+{
+    printf("............. Writing to file: .....................................\n");
+
+    strcat(path, filename);
+    printf("%s \n", path);
+
+    fp=fopen(path, "w");
+
+    /* from the abuf_hexdump() code I'm assuming every element in buffer is of size 32.. */
+    size_t bytes_written = fwrite(buffer, 1, length, fp);
+    printf("bytes_written: %d; length: %d\n", bytes_written, length);
+    assert(bytes_written == length);
+    fclose(fp);
+    printf("............. Done. ................................................\n");
+}
+
+static void aodv_test_writer_write_new_rreq(void)
+{
+
+    strcpy(filename, "plain_rreq");
+    strcpy(path, "./assets/");
+
+    aodv_send_rreq(&plain_rreq);
+}
+
+static void aodv_test_writer_write_more_recent_rreq(void)
+{
+    plain_rreq.origNode.seqnum += 1;
+
+    strcpy(filename, MORE_RECENT_RREQ);
+    strcpy(path, "./assets/");
+
+    aodv_send_rreq(&plain_rreq);
+}
+
+static void aodv_test_writer_write_new_rrep(void)
+{
+
+    strcpy(filename, PLAIN_RREP);
+    strcpy(path, "./assets/");
+
+    aodv_send_rrep(&plain_rrep, &aodvv2_test_sender_ta);
+}
+
+static void aodv_test_writer_write_more_recent_rrep(void)
+{
+    plain_rreq.targNode.seqnum += 1;
+
+    strcpy(filename, MORE_RECENT_RREP);
+    strcpy(path, "./assets/");
+
+    aodv_send_rrep(&plain_rrep, &aodvv2_test_sender_ta);
+}
+
+/* Store packets in files so that we can use them cleanly for testing */
+void write_packets_to_files(void)
+{
+    printf("============= Preparing to write packets to files ==================\n");
+
+    /* Make sure the threads are up and running */
+    sleep(2);
+
+    if (0 != aodv_test_writer_init_data()){
+        printf ("FAILED: unable to init data!\n");
+        return;
+    }
+    /* overwrite the aodvv2 packet writer */
+    aodv_packet_writer_init(aodv_test_write_packet);
+
+    aodv_test_writer_write_new_rreq();
+
+    /* make sure sender_thread is done */
+    sleep(2);
+    aodv_test_writer_write_more_recent_rreq();
+
+    /* make sure sender_thread is done */
+    sleep(2);
+    aodv_test_writer_write_new_rrep();
+
+    /* make sure sender_thread is done */
+    sleep(2);
+    aodv_test_writer_write_more_recent_rrep();
```

Comment:
Weird, this works on my machine too. I'd suggest this is because the call to ``aodv_init()`` that comes after it overwrites the callback too quickly; I'll add another sleep(2), can you test if it works for you too then?